src/targets.c

```diff
@@ -94,14 +94,13 @@ int check_linked_list(struct target *list) {
 }
 
 void destroy_linked_list(struct target *list) {
-	struct target *next = list;
-
-	// jump to the end of the linked list
-	while(next) {
-		list = next;
-		next = list->next;
-		list->next = NULL; // invalidate the pointer because my shit code
-		free(list);
+	if(list->next) {
+		destroy_linked_list(list->next);
 	}
+	//ip_s and mac_s need a seperate free call because they was malloced with sscanf
+	free(list->ip_s);
+	free(list->mac_s);
+	free(list);
+	list = NULL; //invalidate the pointer
```

Comment:
Does this invalidate parent list->next too? It is the same memory pointer, but don't know if both pointers to the same address get invalidated.

But maybe parent  list->next don't need to be invalidated because I invalidate the parent list